README.md

```diff
@@ -16,6 +16,18 @@ A possible outcome would be an emulator that can run RV32I assembly.
   - follow instructions on https://nixos.wiki/wiki/Flakes#Enable_flakes
 - (Optional) install [`direnv`](https://direnv.net)
 
+# Project structure
+- The structure of the project is dictated by the tools we are using, namely Nix and Cabal. This repository is both a Nix flake and a Cabal package and as such contains the following folders/files:
+  - [`app/`](/app): Contains Haskell source for an executable that can be built and run by `cabal run`. It does not do anything meaningful at the moment.
```

Comment:
```suggestion
  - [`app/`](/app): Haskell source code for an executable that can be built and run by `cabal run`.
```

It does not do anything right now, but we'd have to update it once it does, and I'll bet we'd forget that.